components/style/gecko/wrapper.rs

```diff
@@ -92,6 +92,26 @@ use string_cache::{Atom, Namespace, WeakAtom, WeakNamespace};
 use stylesheets::UrlExtraData;
 use stylist::Stylist;
 
+/// A simple wrapper over `nsIDocument`.
+#[derive(Clone, Copy)]
+pub struct GeckoDocument<'ld>(pub &'ld structs::nsIDocument);
```

Comment:
Would it also make sense to implement Deref on this type so you don't need to .0 everywhere?

Reply:
Nah, I'd prefer to be explicit, we don't implement `Deref` on any of them.